src/lib/analytics/segmentOneTrustIntegration.ts

```diff
@@ -0,0 +1,258 @@
+    return window.OnetrustActiveGroups
+  } else {
+    return ""
+  }
+}
+
+function oneTrustReady() {
+  // OneTrust is ready when window.OnetrustActiveGroups contains at least C0001.
+  if (
+    typeof window.OnetrustActiveGroups === "string" &&
+    window.OnetrustActiveGroups.split(",").includes("C0001")
+  ) {
+    return true
+  }
+  return false
+}
+
+function delay(n) {
+  return new Promise<void>(done => {
+    setTimeout(() => {
+      done()
+    }, n)
+  })
+}
+
+async function fetchDestinations(writeKey) {
+  let destinations = await fetchDestinationForWriteKey(writeKey)
+
+  destinations = [
+    ...destinations
+      .reduce((map, item) => {
+        if (item.id === "Repeater") return map // remove Segment Repeater destinations
+        map.has(item["id"]) || map.set(item["id"], item)
+        return map
+      }, new Map()) // return object
+      .values(),
+  ]
+
+  return destinations
+}
+
+function setSegmentDestinationPref(consent, destinations) {
+  // map Segment destination id to OneTrust cookie category.
+  const segmentToOneTrust = {
+    AdWords: "C0004",
+    "Amazon S3": "C0001",
+    Appboy: "C0001",
+    "Facebook Pixel": "C0004",
+    "Google Analytics": "C0002",
+    Indicative: "C0002",
+    "Segment.io": "C0001",
+  }
+
+  const consentArray = consent.split(",")
+
+  const basePref = {
+    // default to false for all Segment destinations.
+    // so if we don't set preference for a destination, it will not be loaded.
+    All: false,
+    // whether we send events to Segment itself.
+    "Segment.io": consentArray.some(d => d === SEGMENT_MAPPING),
+  }
+
+  // for each destination, if its id maps to a OneTrust category that is present in consent, set it true, to be enabled.
+  const additionalPref = destinations
+    .map(function (dest) {
+      if (
+        dest.id in segmentToOneTrust &&
+        consentArray.includes(segmentToOneTrust[dest.id])
+      ) {
+        return { [dest.id]: true }
+      } else {
+        // no mapping for the destination, or it maps to a OneTrust category that is not present in OneTrust consent.
+        return { [dest.id]: false }
+      }
+    })
+    .reduce((acc, val) => {
+      return {
+        ...val,
+        ...acc,
+      }
+    }, [])
+
+  const destinationPref = {
+    ...basePref,
+    ...additionalPref,
+  }
+
+  return destinationPref
+}
+
+function conditionallyLoadAnalytics({
+  writeKey,
+  destinationPref,
+  shouldReload = true,
+}) {
+  let isAnythingEnabled = false
+
+  for (const destination in destinationPref) {
+    const isEnabled = destinationPref[destination]
+    if (isEnabled) {
+      isAnythingEnabled = true
+    }
+  }
+
+  // @ts-ignore
+  if (window.analytics.initialized) {
+    // Segment has been initialized, a page reload is required to reload Segment.
+    if (shouldReload) {
+      window.location.reload()
+    }
+    return
+  }
+
+  if (isAnythingEnabled) {
+    // @ts-ignore
```

Comment:
putting 
```
declare const window: any
```
at top of file will fix all of these

Reply:
done. thanks!